src/liveEditor/__test__/visualEditorClick.test.ts

```diff
@@ -0,0 +1,3160 @@
+                LiveEditorPostMessageEvents.FOCUS_FIELD,
+                {
+                    DOMEditStack: getDOMEditStack(container),
+                }
+            );
+        });
+    });
+
+    describe("HTML RTE field", () => {
+        let htmlRteField: HTMLParagraphElement;
+        let visualEditor: VisualEditor;
+
+        beforeEach(() => {
+            htmlRteField = document.createElement("p");
+            htmlRteField.setAttribute(
+                "data-cslp",
+                "all_fields.bltapikey.en-us.rich_text_editor"
+            );
+            document.body.appendChild(htmlRteField);
+            visualEditor = new VisualEditor();
+        });
+
+        afterEach(() => {
+            visualEditor.destroy();
+        });
+
+        test("should have outline", () => {
+            htmlRteField.dispatchEvent(mouseClickEvent);
+            expect(htmlRteField.classList.contains("cslp-edit-mode"));
+        });
+
+        test("should have an overlay", () => {
+            htmlRteField.dispatchEvent(mouseClickEvent);
+            const overlay = document.querySelector(".visual-editor__overlay");
+            expect(overlay!.classList.contains("visible"));
+        });
+
+        test("should have a field path dropdown", () => {
+            htmlRteField.dispatchEvent(mouseClickEvent);
+            const toolbar = document.querySelector(
+                ".visual-editor__focused-toolbar__field-label-wrapper__current-field"
+            );
+            expect(toolbar).toBeInTheDocument();
+        });
+
+        test("should contain a data-cslp-field-type attribute", async () => {
+            VisualEditor.VisualEditorGlobalState.value.previousSelectedEditableDOM =
+                htmlRteField;
+
+            await waitFor(() => {
+                htmlRteField.dispatchEvent(mouseClickEvent);
+            });
+
+            expect(htmlRteField).toHaveAttribute(
+                LIVE_EDITOR_FIELD_TYPE_ATTRIBUTE_KEY
+            );
+        });
+
+        test("should not contain a contenteditable attribute", async () => {
+            VisualEditor.VisualEditorGlobalState.value.previousSelectedEditableDOM =
+                htmlRteField;
+
+            await waitFor(() => {
+                htmlRteField.dispatchEvent(mouseClickEvent);
+            });
+
+            expect(htmlRteField).not.toHaveAttribute("contenteditable");
+        });
+
+        test("should send a focus field message to parent", async () => {
+            VisualEditor.VisualEditorGlobalState.value.previousSelectedEditableDOM =
+                htmlRteField;
+
+            await waitFor(() => {
+                htmlRteField.dispatchEvent(mouseClickEvent);
+            });
+
+            expect(liveEditorPostMessage?.send).toBeCalledWith(
+                LiveEditorPostMessageEvents.FOCUS_FIELD,
+                {
+                    DOMEditStack: getDOMEditStack(htmlRteField),
+                }
+            );
+        });
```

Comment:
Does this test case work properly? We didn't see the frame open for the RTE, so this should fail, isn't it?

Reply:
The message is sent properly to the parent : visual-editor. The issue is because of `RESPONSE` being improper from the parent.